Dockerfile

```
@@ -0,0 +1,57 @@
+# Multi-stage build for crawler_mcp with integrated webhook server
+FROM python:3.11-slim as builder
+
+WORKDIR /app
+
+# Copy project files
+COPY pyproject.toml uv.lock ./
+
+# Install uv and sync dependencies
+RUN pip install --no-cache-dir uv && \
```

Hadolint (2.12.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)